### PR TITLE
Add recaptcha

### LINKED ISSUE #541: Add reCAPTCHA to 'register' page
Add an additional validation on 'register' page which will permit bots to register accounts.
I suggest using https://www.google.com/recaptcha/intro/

### PR DESCRIPTION
**Pull request by denys-medynskyi**
Resolves #541

### Description

Adds Google recaptcha on user sign up form.
To manual test it you need tokens. I have them if you need.
No specs 😿

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -32,5 +35,23 @@ def sign_up_params
...
+      end
+    end
+
+    def add_recaptcha_error
+      @recaptcha_error_message = flash[:recaptcha_error]
+      flash.delete(:recaptcha_error)
+    end
+
+    def recaptcha_error_message
+      @recaptcha_error_message
```
Comment (sveredyuk) [inline app/controllers/users/registrations_controller.rb]: could be simple attr_reader

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 module Users
   class RegistrationsController < Devise::RegistrationsController
+    prepend_before_action :check_captcha, only: [:create]
```
Comment (sveredyuk) [inline app/controllers/users/registrations_controller.rb]: `only: :create`

**Review by sveredyuk — DISMISSED** [review]

File: app/controllers/users/registrations_controller.rb [hunk 3]
```diff
@@ -32,5 +35,23 @@ def sign_up_params
         :subscribed
       )
     end
+
+    def check_captcha
+      unless verify_recaptcha
```
Comment (floor114) [inline app/controllers/users/registrations_controller.rb]: `return if verify_recaptcha`

**Review by floor114 — DISMISSED** [review]

**Comment by denys-medynskyi** [comment]
@sveredyuk @floor114 @kushniryb please take a look again. 
Also added recaptcha to the forgot password form, so I extracted shared thing to the module.

**Review by floor114 — APPROVED** [review]

**Review by sveredyuk — APPROVED** [review]

**Review by kushniryb — APPROVED** [review]

**Pull request merged**